08_AdvancedTypes/Program.cs

```diff
@@ -0,0 +1,108 @@
+﻿using _08_AdvancedTypes;
+using _08_AdvancedTypes.DTO;
+using System.Text.Json;
+
+IApiDataReader starwarsAPI = new StarWarsApiDataReader();
+
+string responseContent = await starwarsAPI.ReadAsync("https://swapi.dev/api/", "planets");
+
+IEnumerable<Planet> planets =
+    JsonSerializer.Deserialize<PlanetsDTO>(responseContent).
+    results.Select(p =>
+     new Planet()
+     {
+         Name = p.name,
+         Diameter = int.TryParse(p.diameter, out int res) ? res + "" : "",
+         SurfaceWater = int.TryParse(p.surface_water, out res) ? res + "" : "",
+         Population = int.TryParse(p.population, out res) ? res + "" : ""
+     });
+
+var tablePrinter = new UniversalTablePrinter(planets);
+tablePrinter.PrintToConsole();
+
+
+Console.WriteLine("The statistics of which property would you like to see?" +
+    $"{Environment.NewLine}population" +
+    $"{Environment.NewLine}diameter" +
+    $"{Environment.NewLine}surface water");
+
+
+string desiredStat = Console.ReadLine().ToLower();
+switch (desiredStat)
+{
+    case "population":
+        Planet pMin = planets.First();
+        Planet pMax = planets.First();
+        foreach (Planet planet in planets)
+        {
+            int.TryParse(planet.Population, out int currentValue);
```

Comment:
You now have to `TryParse` for each planet again because you didn't save the `int` values during conversion